nupic/research/frameworks/dendrites/model_utils.py

```diff
@@ -106,12 +111,15 @@ def train_dendrite_model(
         if share_labels:
             target = target % num_labels
 
-        data = data.to(device)
-        target = target.to(device)
         if context is not None:
-            context = context.to(device)
+            context = context.to(device, non_blocking=async_gpu)
+
+        # FIXME: Pytorch 1.7: Replace with optimizer.zero_grad(set_to_none=True)
+        # optimizer.zero_grad(set_to_none=True)
+        for group in optimizer.param_groups:
+            for p in group["params"]:
```

Comment:
`optimizer.zero_grad(set_to_none=True) ` is not available in torch 1.6.

Reply:
The original `optimizer.zero_grad` will call `p.grad.zero_()`  using more memory and also performing the optimizer step on zero gradients. The optimizer usually skips the step if the gradient is `None` but it will not skip if the gradient is zero
see
- https://pytorch.org/tutorials/recipes/recipes/tuning_guide.html#use-parameter-grad-none-instead-of-model-zero-grad-or-optimizer-zero-grad 
- https://pytorch.org/docs/master/generated/torch.optim.Optimizer.zero_grad.html#torch.optim.Optimizer.zero_grad